### PR TITLE
Expose UCT performance info via iface_query and pd_query

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1341/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/732/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
autobot build

**Comment by shamisp** [comment]
@yosefe , why it is marked as invalid ?

**Comment by yosefe** [comment]
I want to rebase it on master

**Comment by yosefe** [comment]
@shamisp pls take a look

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/762/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1373/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/763/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1374/ for details.

**Comment by shamisp** [comment]
how did you come up with the original numbers ? @MattBBaker please take a look as well.

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -221,6 +234,10 @@ struct uct_iface_attr {
 
     size_t                   iface_addr_len; /**< Size of interface address */
     size_t                   ep_addr_len;    /**< Size of endpoint address */
+
+    double                   overhead;     /**< Message overhead, seconds */
```
Comment (shamisp) [inline src/uct/api/uct.h]: what is the message overhead ? can you please give an example

**Comment by yosefe** [comment]
@shamisp I copied shared memory and ugni numbers from existing code and added estimation for "overhead".

File: src/uct/api/uct.h (hunk 1 above)
Comment (yosefe) [inline src/uct/api/uct.h]: it should be the overhead of every message, which limits the message rate and also added to latency

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1375/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/764/ for details (Mellanox internal link).

File: src/uct/api/uct.h (hunk 1 above)
Comment (shamisp) [inline src/uct/api/uct.h]: Well, it is most like to be runtime dependent (cpu architecture, etc )  ?

Comment (yosefe) [inline src/uct/api/uct.h]: yes, for now it is constant, later on it should be calculated in runtime

**Comment by miked-mellanox** [comment]
:+1:

**Comment by miked-mellanox** [comment]
@yosefe - need rebase, merge conflict

File: src/uct/api/uct.h (hunk 1 above)
Comment (shamisp) [inline src/uct/api/uct.h]: Please add more documentation for the parameter describing the goal, and the fact that it can be architecture depended.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1390/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/778/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/779/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1391/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/782/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1394/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1396/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/784/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask i've fixed dev_type

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1399/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/786/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/787/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1400/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1401/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/788/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1406/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/792/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/798/ for details (Mellanox internal link).

**Pull request merged**